### PR TITLE
Fill gaps in manager coverage

### PR DESCRIPTION
**Pull request by marcofavorito**
## Proposed changes

Fill coverage gap in manager due to #2523 

## Fixes

n/a

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [X] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

n/a

### CODE REVIEW

File: aea/manager/manager.py [hunk 1]
```diff
@@ -366,6 +366,7 @@ def stop_manager(
             self._thread.join(self.DEFAULT_TIMEOUT_FOR_BLOCKING_OPERATIONS)
 
         self._thread = None
+        self._warning_message_printed_for_agent = {}
```
Comment (marcofavorito) [inline aea/manager/manager.py:369]: I feel this was missing? Whenever we stop the multiagent manager, we want that the next time it is run, the warning messages are handled again

File: tests/test_manager/test_manager.py [hunk 2]
```diff
@@ -284,12 +284,12 @@ def test_default_exception_handling(self, *args):
         assert behaviour
 
         with patch.object(
-            self.manager, "_print_exception_occurred_but_no_error_callback"
+            self.manager, "_print_exception_occurred_but_no_error_callback",
+            side_effect=self.manager._print_exception_occurred_but_no_error_callback
         ) as callback_mock:
             with patch.object(behaviour, "act", side_effect=ValueError("expected")):
-                self.manager.start_all_agents()
```
Comment (marcofavorito) [inline tests/test_manager/test_manager.py:290]: not needed as agents are already started above

File: tests/test_manager/test_manager.py [hunk 3]
```diff
@@ -284,12 +284,12 @@ def test_default_exception_handling(self, *args):
         assert behaviour
 
         with patch.object(
-            self.manager, "_print_exception_occurred_but_no_error_callback"
+            self.manager, "_print_exception_occurred_but_no_error_callback",
+            side_effect=self.manager._print_exception_occurred_but_no_error_callback
```
Comment (marcofavorito) [inline tests/test_manager/test_manager.py:288]: added side_effect so we both count the calls and execute the function

File: aea/manager/manager.py [hunk 4]
```diff
@@ -874,7 +875,7 @@ def _print_exception_occurred_but_no_error_callback(
         :param exception: the caught exception.
         """
         if self._warning_message_printed_for_agent.get(agent_name, False):
-            return
+            return  # nocover
```
Comment (DavidMinarsch) [inline aea/manager/manager.py:878]:
```suggestion
            return  # pragma: nocover
```

**Pull request merged**